content/concepts/consensus-network/transaction-malleability.md

```diff
@@ -0,0 +1,142 @@
+- Be prepared for the possibility that a transaction executes with a different hash and set of signatures than you expected. Carefully monitor your account's sent transactions (for example, using the [account_tx method][]).
+- If you re-create transactions to be multi-signed, _do not_ change the `Sequence` number unless you have manually confirmed that the intended actions have not already executed.
+- Confirm that the tfFullyCanonicalSig flag is enabled before signing.
+
+For greater security, these guidelines provide multiple layers of protection.
+
+
+## Exploit With Malleable Transactions
+
+If the software you use to interface with the XRP Ledger sends malleable transactions, a malicious actor may be able to trick your software into losing track of a transaction's final outcome and potentially (in the worst case) sending equivalent payments multiple times.
+
+If you use single-signatures and always enable the tfFullyCanonicalSig flag, you are not vulnerable to this exploit. If you use multi-signatures, you may be vulnerable if you or your signers provide more signatures than necessary.
+
+### Exploit Scenario Steps
+
+The process to exploit a vulnerable system follows a series of steps similar to the following:
+
+1. The vulnerable system constructs and signs a transaction without enabling tfFullyCanonicalSig.
+
+    Three ways that a transaction may not enable the tfFullyCanonicalSig flag are:
+
+    - The system explicitly specifies a `Flags` field that does not have the tfFullyCanonicalSig bit enabled.
+    - The transaction is multi-signed and does not explicitly enable the tfFullyCanonicalSig flag.
+    - The system omits the `Flags` field from the transaction fields, but uses a non-standard implementation that does not automatically enable tfFullyCanonicalSig when signing.
+
+    To be vulnerable, the transaction must be signed with an ECDSA key pair. If multi-signed, the transaction must be signed by at least one ECDSA key pair.
+
+    Most likely, the vulnerable transaction uses a fully-validated signature, but the flags indicate that the transaction would also be valid with a non-fully-canonical one. The transaction may also use `LastLedgerSequence` so that its final outcome is clear in a finite amount of time.
+
+2. The system notes the identifying hash of the vulnerable transaction, submits it to the XRP Ledger network, then begins monitoring for that hash to be included in a validated ledger version.
+
+3. A malicious actor sees the transaction propagating through the network before it becomes confirmed.
+
+4. The malicious actor calculates the alternate signature for the vulnerable transaction, and derives the hash of the resulting transaction data.
+
+    Unlike creating a signature for a different transaction data, this does not require a large amount of computational work. It can be done in much less time than it takes to generate a signature in the first place.
```

Comment:
I'm not a cryptographer, so I'm assuming you are correct -- that the alternative signature is relatively easy to compute given only the original transaction with the original signature.  @nbougalis, could you confirm?  Thanks.

Reply:
My understanding is that the "work required" is literally subtracting an integer mod another integer (so the description here may be an understatement!) but I didn't test it myself so I can only go on what I read elsewhere.

Reply:
nit: possible edit?

>Unlike creating a signature for a different transaction data, this 

to

>Unlike creating a signature for a different transaction, this

Reply:
Went slightly the other direction with the edit:

> Unlike creating a signature for different transaction instructions, this